src/Elastic.Channels/BufferedChannelBase.cs

```diff
@@ -199,26 +206,34 @@ public async Task<bool> WaitToWriteManyAsync(IEnumerable<TEvent> events, Cancell
 	/// <inheritdoc cref="IBufferedChannel{TEvent}.TryWriteMany"/>
 	public bool TryWriteMany(IEnumerable<TEvent> events)
 	{
-		var written = true;
+		var allWritten = true;
 
 		foreach (var @event in events)
 		{
-			written = TryWrite(@event);
+			var written = TryWrite(@event);
+			if (!written) allWritten = written;
 		}
 
-		return written;
+		return allWritten;
 	}
 
 	/// <inheritdoc cref="ChannelWriter{T}.WaitToWriteAsync"/>
 	public virtual async Task<bool> WaitToWriteAsync(TEvent item, CancellationToken ctx = default)
 	{
 		ctx = ctx == default ? TokenSource.Token : ctx;
+
+		// small interop to protect inbound task from cpu stealing
+		if (_outstandingOperations % BufferOptions.OutboundBufferMaxSize / 10 == 0)
```

Comment:
This is a fix for cases like `examples/Elastic.Channels.Continuous` that overproduce using all threads in a very tight loop and effectively thread steal where the outbound tasks can never catch up.